src/edu/uw/easysrl/syntax/parser/ParserBuilder.java

```diff
@@ -30,24 +30,12 @@
 public abstract class ParserBuilder<T extends ParserBuilder<T>> {
 
 	ParserBuilder(final File modelFolder) {
-		this.modelFolder = modelFolder;
-		this.jointModel = new File(modelFolder, "weights").exists();
-		try {
-			this.unaryRules = AbstractParser.loadUnaryRules(new File(modelFolder, "unaryRules"));
-			this.lexicalCategories = TaggerEmbeddings.loadCategories(new File(modelFolder, "categories"));
-			this.seenRules = new SeenRules(new File(modelFolder, "seenRules"), lexicalCategories);
-			final File cutoffsFile = new File(modelFolder, "cutoffs");
-			cutoffs = cutoffsFile.exists() ? Util.deserialize(cutoffsFile) : null;
-			if (new File(modelFolder, "markedup").exists()) {
-				Coindexation.parseMarkedUpFile(new File(modelFolder, "markedup"));
-			}
-		} catch (final IOException e) {
-			throw new UncheckedIOException(e);
-		}
-
+		modelFolder(modelFolder);
 	}
 
-	final File modelFolder;
+	ParserBuilder() { }
```

Comment:
It's so that the deprecated constructors have access to default arguments. See https://github.com/mikelewis0/EasySRL/pull/35/files#diff-4e293fd3951760563ecc6e7c728a4fb3R78

One way to resolve this nicely would be to completely separate model objects from the files in the builder. Reading in a set of objects from a model folder should be additional rather than core functionality. It would also allow for proper unit testing through dependency injection (e.g. providing mock grammars .